src/background/utils/network.ts

```diff
@@ -38,12 +38,18 @@ interface CacheRecord {
 class LimitedCacheStorage {
     static QUOTA_BYTES = ((navigator as any).deviceMemory || 4) * 16 * 1024 * 1024;
     static TTL = getDuration({minutes: 10});
+    static ALARM_NAME = 'network';
 
     private bytesInUse = 0;
     private records = new Map<string, CacheRecord>();
 
     constructor() {
-        setInterval(() => this.removeExpiredRecords(), getDuration({minutes: 1}));
+        chrome.alarms.onAlarm.addListener(async (alarm) => {
+            if (alarm.name === LimitedCacheStorage.ALARM_NAME) {
```

Comment:
Hmm, could I ask why we are checking if the the alarm_name is the same? Is this some kind of intentional design of the alarm API?

Reply:
Alarm name allows us to react only to the desired alarm, which is helpful when we have multiple different classes creating their own alarms for different things. Without name check every alarm listener would react to every alarm. E.g., `Newsmaker` would react to `LimitedCacheStorage` alarm and query your server for news every minute when you actually want it to do it every 4 hours.

Reply:
Aha, that makes more change.